src/inmanta/deploy/scheduler.py

```diff
@@ -162,6 +168,9 @@ def reset(self) -> None:
         self._work.reset()
         self._workers.clear()
         self._deploying_latest.clear()
+        for task in self.resource_periodic_deploys.values():
+            task.cancel()
```

Comment:
Don't you still need to await a task you cancel?
```suggestion
            task.cancel()
        with contextlib.suppress(asyncio.CancelledError):
            await asyncio.gather(*self.resource_periodic_deploys.keys())
```

(not sure if this works but I think it will)